src/common/tree/tree_node_serializer.py

```diff
@@ -225,7 +215,7 @@ def tree_node_from_dict(
             if not isinstance(attribute_data, dict):
                 raise ValidationException(
                     f"The attribute '{child_attribute.name}' on blueprint '{node.type}' "
-                    + f"should be a dict, but was '{str(type(attribute_data))}'"
+                    + f"should be a dict, but was '{type(attribute_data)!s}'"
```

Comment:
Duplicate:

Pretty sure the !s is unecessary here.
Tested this in playground, the !s makes no difference in output.